rs/nervous_system/agent/src/pocketic_impl.rs

```diff
@@ -205,4 +211,35 @@ impl CallCanisters for PocketIc {
     fn caller(&self) -> Result<Principal, Self::Error> {
         Ok(Principal::anonymous())
     }
+
+    fn is_canister_stopped_error(&self, err: &Self::Error) -> bool {
+        match err {
+            PocketIcCallError::PocketIc(err) => {
+                [ErrorCode::CanisterStopped, ErrorCode::CanisterStopping].contains(&err.error_code)
+            }
+            _ => false,
+        }
+    }
+}
+
+impl ProgressNetwork for PocketIcAgent<'_> {
+    async fn progress(&self, duration: Duration) {
+        self.pocket_ic.progress(duration).await
+    }
+}
+
+impl ProgressNetwork for PocketIc {
+    async fn progress(&self, duration: Duration) {
+        if !self.auto_progress_enabled().await {
+            self.advance_time(duration).await;
+            self.tick().await;
+        } else {
+            // Otherwise, we have to wait for the time to pass "naturally".
+            if duration > Duration::from_secs(5) {
```

Comment:
In addition to this check, we may want to panic if the suggested wait is longer than 10 minutes or something like that.  We really don't want tests on CI to take that long?

Reply:
Tests on CI are supposedly run on non-live PocketIC and hence they won't actually call `tokio::time::sleep(duration)`.

I'm not aware of any CI jobs that run using an `ic-agent` backend. If there are some, then panicking would make sense